### ISSUE TITLE
Rule proposal: `no-class-list-add-add`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#2160]
### Description

Like [`no-array-push-push`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-push-push.md) but for [`DOMTokenList.add`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/add).

### Fail

```js
el.classList.add('foo');
el.classList.add('bar', 'baz');
```

### Pass

```js
el.classList.add('foo', 'bar', 'baz');
```

### Additional Info

Maybe also name it `no-classlist-add-add` because the rule wil likely match on that string and does not actually know the type.

**Comment by silverwind** [comment]
Renamed to `no-classlist-add-add`, I think it's clearly a better name.

**Comment by silverwind** [comment]
I think I could have a try at implementing this rule.

**Comment by B2o5T** [comment]
`classList.remove` accepts also multiple arguments so that we can have one generic rule for both `.add` and `.remove` methods

## Fail
```js
el.classList.remove('foo');
el.classList.remove('bar', 'baz');
```
## Pass
```js
el.classList.remove('foo', 'bar', 'baz');
```

**Comment by silverwind** [comment]
Any name suggestions when covering remove as well? `no-classlist-repeated-modify`?

**Comment by fisker** [comment]
We should add a dash to `classlist`.

**Comment by silverwind** [comment]
I'm thinking of an even broader name: `prefer-combinable-action`. It could have a in-rule config like:

```js
const matches = [
  ['classList', 'add'],
  ['classList', 'remove'],
  ['push'],
];
```

This would be very easy to extend later on and it could maybe also replace `no-array-push-push`.

**Comment by fisker** [comment]
I was thinking the same easier, so I searched the JavaScript spec easier today, except `Math.{min,max}()` there seems no other methods accepts unlimited arguments, and `Math.{min,max}()` normally assigned to a variable not exactly the same as `Array#{push,unshift}()`. Then I give up the idea to combine them into one rule.

**Comment by silverwind** [comment]
Thanks for searching. I was almost sure that [`Set.add`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add) would support it as well, but alas, no.

**Comment by silverwind** [comment]
> We should add a dash to `classlist`.

I take it this is consistent with other rule names?

**Comment by B2o5T** [comment]
> I'm thinking of an even broader name: `prefer-combinable-action`. It could have a in-rule config like:
> 
> ```js
> const matches = [
>   ['classList', 'add'],
>   ['classList', 'remove'],
>   ['push'],
> ];
> ```
> 
> This would be very easy to extend later on and it could maybe also replace `no-array-push-push`.

interested but for adding `Arrray#unshift` it should suggest inverted order.

```js
const matches = [
  ['classList', 'add'],
  ['classList', 'remove'],
  ['push'],
  { path: ['unshift'], invertOrder: true }
]
```

## Fail
```js
foo.unshift(1);
foo.unshift(2);
    ^^^^^^^ Do not call `#unshift()` multiple times. Argument order should be inverted!
foo.unshift(3);
    ^^^^^^^ Do not call `#unshift()` multiple times. Argument order should be inverted!
```

## Suggestion
```js
foo.unshift(3, 2, 1);
```

**Comment by sindresorhus** [comment]
Accepted as `no-class-list-add-add`

**Comment by axetroy** [comment]
When I tried to apply this rule to my code base, I found it less readable (even though there were a few fewer method calls)

**Comment by fisker** [comment]
[`importScripts`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts) accept mutiple arguments too.

**Comment by sindresorhus** [comment]
Should be `prefer-combined-actions` (plural is correct in this case).

**Comment by sindresorhus** [comment]
But how about `prefer-single-call`? That captures the intend, to use a single call instead of multiple calls, whenever possible.
